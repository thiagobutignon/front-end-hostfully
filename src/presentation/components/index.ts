export { default as ErrorComponent } from './error/error'
export { default as HeaderComponent } from './header/header'
export { default as InputError } from './input/input-error'
export { default as InputWithLabel } from './input/input-with-label'
export { default as LayoutComponent } from './layout/layout'
export { default as PropertyInfoComponent } from './property-info/property-info'
export { default as PropertyCardComponent } from './select/property-select'
export { default as ThemeModeComponent } from './theme-mode/theme-mode'
