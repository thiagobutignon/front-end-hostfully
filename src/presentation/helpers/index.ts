export * from './populate-form-fields'
export * from './wrap-chakra'
