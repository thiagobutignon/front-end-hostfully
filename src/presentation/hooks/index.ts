export * from './booking'
export * from './error'
export * from './property'
