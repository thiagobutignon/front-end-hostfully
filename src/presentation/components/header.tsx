import { Box, Flex, Menu, Stack, Text } from '@chakra-ui/react'

import { ThemeModeComponent } from '@/presentation/components'

const HeaderComponent: React.FC = () => {
  return (
    <>
      <Box bg="reverseBackground" px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Text color="reverseText" fontSize="xl" fontWeight={700}>
            Hostfully
          </Text>

          <Flex alignItems={'center'}>
            <Stack direction={'row'}>
              <Menu>
                <ThemeModeComponent />
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  )
}

export default HeaderComponent
