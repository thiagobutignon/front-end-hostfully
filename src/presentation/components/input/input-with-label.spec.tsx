import { RenderResult, fireEvent, render } from '@testing-library/react'

import { InputWithLabel } from '@/presentation/components'

describe('InputWithLabel', () => {
  const initialState = {
    myInput: '',
    myInputError: 'Error message'
  }
  const setState = jest.fn()
  const makeSut = (state: any = initialState): RenderResult => {
    return render(
      <InputWithLabel
        type="text"
        state={state}
        setState={setState}
        name="myInput"
      >
        My Label
      </InputWithLabel>
    )
  }

  test('should render with label and input', () => {
    const { getByTestId, getByRole } = makeSut()
    expect(getByTestId('myInput-label')).toHaveTextContent('My Label')
    expect(getByRole('textbox')).toBeInTheDocument()
  })

  test('should call setState on input change', () => {
    const { getByRole } = makeSut()
    const input = getByRole('textbox')
    fireEvent.change(input, { target: { value: 'new value' } })
    expect(setState).toHaveBeenCalledWith({
      ...initialState,
      myInput: 'new value'
    })
  })

  test('should display error if present in state', () => {
    const errorState = {
      myInput: '',
      myInputError: 'Error message'
    }
    const { getByTestId } = makeSut(errorState)
    const errorMessage = getByTestId('myInput-error')
    expect(errorMessage).toHaveTextContent('Error message')
  })

  test('matches snapshot', () => {
    const { asFragment } = render(
      <InputWithLabel
        type="text"
        state={initialState}
        setState={setState}
        name="myInput"
      >
        My Label
      </InputWithLabel>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
