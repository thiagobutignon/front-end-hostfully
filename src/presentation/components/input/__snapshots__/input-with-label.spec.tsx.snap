// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <label
      class="chakra-form__label css-1muhubi"
      data-testid="myInput-label"
      for="field-:r3:"
      id="field-:r3:-label"
    >
      My Label
    </label>
    <input
      aria-describedby="field-:r3:-helptext"
      autocomplete="new-myInput"
      class="chakra-input css-i56gs7"
      data-testid="myInput"
      id="field-:r3:"
      name="myInput"
      type="text"
      value=""
    />
    <div
      class="chakra-form__helper-text css-18w01t4"
      data-testid="myInput-error"
      id="field-:r3:-helptext"
    >
      Error message
    </div>
  </div>
</DocumentFragment>
`;
